src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -340,6 +343,62 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  def updateCluster(userInfo: UserInfo,
+                    googleProject: GoogleProject,
+                    clusterName: ClusterName,
+                    clusterRequest: ClusterRequest): Future[Cluster] = {
+    for {
+      cluster <- internalGetActiveClusterDetails(googleProject, clusterName) //throws 404 if nonexistent
+
+      _ <- checkClusterPermission(userInfo, DeleteCluster, cluster) //throws 404 if no auth //TODO: check a more appropriate action here
+
+      _ <- internalUpdateCluster(cluster, clusterRequest)
+
+      updatedCluster <- internalGetActiveClusterDetails(googleProject, clusterName)
+    } yield { updatedCluster }
+  }
+
+  def internalUpdateCluster(existingCluster: Cluster, clusterRequest: ClusterRequest) = {
+    if(existingCluster.status.isUpdatable) {
```

Comment:
We probably need another conditional for the case where we only update the auto-pause threshold since we wouldn't need to message `ClusterMonitorSupervisor` in that case.

Reply:
We should also update the corresponding test to verify the status of the cluster doesn't change for that case.

Reply:
+1 to this -- if just updating auto pause stuff we probably don't need to change the status to `Updating` and call `ClusterMonitorSupervisor`.